sdk/core/Azure.Core.Amqp/src/AmqpAddress.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Amqp
+{
+    /// <summary>
+    /// Represents the AMQP address.
+    /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-address-string"/>.
+    /// </summary>
+    public struct AmqpAddress : IEquatable<AmqpAddress>
+    {
+        private readonly string _address;
```

Comment:
Would an implicit conversion to `string` make sense?   Seems that will be the majority use case, no?

Reply:
I was thinking of this, but the future extension to this type may be to support URI in which case we would be allocating in the string conversion.